osgp/platform/osgp-adapter-domain-smartmetering/src/main/java/org/opensmartgridplatform/adapter/domain/smartmetering/application/mapping/customconverters/ActualValueConverter.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright 2021 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.domain.smartmetering.application.mapping.customconverters;
+
+import ma.glasnost.orika.CustomConverter;
+import ma.glasnost.orika.MappingContext;
+import ma.glasnost.orika.metadata.Type;
+import org.opensmartgridplatform.domain.core.valueobjects.smartmetering.ActualValue;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ActualValueDto;
+
+public class ActualValueConverter extends CustomConverter<ActualValueDto, ActualValue> {
+
+    @Override
+    public ActualValue convert(final ActualValueDto source,
```

Comment:
can source ever be null? this seems strange. I would expect the data to be validated earlier in the process.

Reply:
Yes, source can be null.